packages/ra-core/src/reducer/admin/ui.ts

```diff
@@ -76,7 +76,7 @@ const uiReducer: Reducer<UIState> = (
         case STOP_OPTIMISTIC_MODE:
             return { ...previousState, optimistic: false };
         default:
-            return previousState;
+            return { ...defaultState, ...previousState };
```

Comment:
This is a big performance issue: Every action will update the UI state, which will cause a redraw of the components depending on it (e.g. sidebar). So this is not an acceptable solution. 

I advise you to check the createAdminStaore instead, to allow partial initialState to be used.

Reply:
The UI initial states are now overridden in the createAdminStore by the partial initialState passed by the user